core/src/test/java/com/datastax/oss/driver/internal/core/type/codec/ListCodecTest.java

```diff
@@ -93,6 +93,12 @@ public void should_decode_non_empty_list() {
         .containsExactly(1, 2, 3);
   }
 
+  @Test
+  public void should_decode_list_with_null_elements() {
+    when(elementCodec.decode(Bytes.fromHexString("0x0002"), ProtocolVersion.DEFAULT)).thenReturn(2);
+    assertThat(decode("0x" + "00000002" + "FFFFFFFF" + "000000020002")).containsExactly(null, 2);
```

Comment:
This is very cryptic, how about:

```
assertThat(
        decode(
            "0x"
                + "00000002" // number of elements
                + "FFFFFFFF" // size of element 1 (null)
                + "00000002" // size of element 2 (2 bytes)
                + "0002" // element 2
            ))
    .containsExactly(null, 2);
```

Reply:
There's a commented example 30 lines above. When you have the whole file open in an IDE it's pretty visible, I think it's fine not to repeat it in every test.

Reply:
I agree with Alex on that, I had the same impression that it was very criptic

Reply:
Well it's not exactly the same example, here we have a `"FFFFFFFF"` block that seems strange at first glance, and `"000000020002"` is also strange since it contains both the size of element 2 and element 2 itself.